### PR TITLE
[workspace] Better hide ccd and fcl externals

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
Fix `ccd` BUILD file to disallow things like `#include "polytope.h"` which previously resolved to its `src` directory.

Remove unnecessary fcl version rules; use strict genrule checking.

Towards #17231.

- [x] Requires #18677 first.

- - -
This change is [Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/18678)

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
+@SeanCurtis-TRI for feature review, please.

I'm hoping the build system changes are clear enough for a general developer audience.  The comments in the `drake/tools/workspace/vendor_cxx.bzl` macro might help.  We can always bring in @rpoyner-tri for build system review if it's too complicated, though.

**Review by SeanCurtis-TRI — COMMENTED** [review]
As a representative of the "general developer audience", I think I can say that the build machinations could be followed but not vetted. I was able to tell myself a story about everything I saw that placed it all in a coherent narrative, but I couldn't generate it on my own and doubt that I could make assertions about correctness. There are enough voodoo bazel rules being used with semantics that aren't clear from the invocation.

So, in conclusion, I think the general developer audience could follow this. But they couldn't validate nor generate it.

:LGTM:

> Reviewed 19 of 23 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/18678#-:-NNHfxO92gBlufHqM3W6:bpvj41g)* status: needs at least two assigned reviewers (waiting on @jwnimmer-tri)

**Review by jwnimmer-tri — COMMENTED** [review]
I suppose the most important thing is if a developer gets a build error, how able would they be to fix it on their own?  Like if they forgot a `DRAKE_NO_EXPORT` and saw an attribute error.  Ah well, I guess we'll find out.

+@sammy-tri for platform review per schedule, please.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/18678#-:-NNIl9CQ8uKhODcOLdlr:b-4a9mek)* status: LGTM missing from assignee sammy-tri(platform) (waiting on @sammy-tri)

**Review by SeanCurtis-TRI — COMMENTED** [review]
Yep. I subscribe to the "we'll find out" as we go. It'll be educational. And, quite simply, it's not like fcl is widely referenced.  

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/18678#-:-NNIxbb97GTvDt1UXruA:b-hyuhk6)* status: LGTM missing from assignee sammy-tri(platform) (waiting on @sammy-tri)

**Review by sammy-tri — COMMENTED** [review]
:lgtm:

> Reviewed 19 of 23 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/18678#-:-NNbsdVQ9p3yB5Qa-WqI:bnfp4nl)* status: :shipit: complete! all discussions resolved, LGTM from assignees sammy-tri(platform),SeanCurtis-TRI(platform) (waiting on @jwnimmer-tri)
